air/src/air/mod.rs

```diff
@@ -357,13 +381,22 @@ pub trait Air: Send + Sync {
     fn get_boundary_constraints<E: FieldElement<BaseField = Self::BaseField>>(
         &self,
         aux_rand_elements: &AuxTraceRandElements<E>,
-        composition_coefficients: &[E],
+        boundary_composition_coefficients: &[E],
+        lagrange_kernel_boundary_coefficient: Option<E>,
     ) -> BoundaryConstraints<E> {
```

Comment:
Do we need to integrate Lagrange kernel boundary constraints into other boundary constraints? In my initial thinking, we'd treat all Lagrange constraints separately. Specifically, instead of having 2 types of constraints as we currently do, we'd have:

1. Transition constraints (same as now).
2. Boundary constraints (same as now).
3. Lagrange kernel constraints - this would include boundary and transition constraints relevant to the Lagrange column.

Reply:
Similar to other comments, I think we should merge them with `TransitionConstraints` and `BoundaryConstraints` (https://github.com/facebook/winterfell/issues/261)

Reply:
I think trying to merge them into current `TransitionConstraints` and `BoundaryConstraints` will result in more complexity than required. It is also not clear to me that the strategies we use for evaluating (and optimizing) transition/boundary constraints will be applicable to Lagrange kernel constraints. So, as mention in the previous comment, my preference would be to keep Lagrange kernel constraints as encapsulated as possible and not treat them as special cases of boundary/transition constrains.

Of course, if later on we decided that it is a better approach, we can always refactor the codebase - but for now, I think it would be good to implement end-to-end handling of Lagrange column separately.

Concretely, we could add a method to the `Air` trait which would look something like this:
```Rust
fn get_lagrange_kernel_constraints<E: FieldElement<BaseField = Self::BaseField>>(
    &self,
    rand_elements: &Vec<E>, // these would be the r-values
    composition_coefficients: &LagrangeConstraintsCompositionCoefficients<E>,
) -> Option<LagrangeKernelConstraints<E>>
```
This method would return `None` if there is no Lagrange kernel column in the trace, and `LagrangeKernelConstraints` otherwise. `LagrangeKernelConstraints` would have something like `evaluate_at()` method, which would evaluate all Lagrange kernel constraints (boundary and transition) for a given frame. (it could also have other methods which would help the prover to perform evaluations in a more optimal way).

Reply:
I had separated them already; I had forgotten to update the `BoundaryConstraints` structs (the Lagrange constraints part was not actually used anymore). Now fixed.

The `LagrangeKernelConstraints` part of the comment still needs to be addressed though.